catroid/src/androidTest/java/org/catrobat/catroid/test/common/DeviceVariableAccessorTest.java

```diff
@@ -0,0 +1,180 @@
+
+		UserVariable result = project1.getUserVariable(userVariable.getName());
+		accessor = new DeviceVariableAccessor(project1.getDirectory());
+		accessor.readUserVariableValue(result);
+
+		assertEquals(result.getDeviceValueKey(), userVariable1.getDeviceValueKey());
+		assertEquals(result.getValue(), userVariable1.getValue());
+	}
+
+	@Test
+	public void copyProjectTest() throws IOException, LoadingProjectException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+		accessor.writeVariable(userVariable);
+		XstreamSerializer.getInstance().saveProject(project);
+
+		Project project2 = new Project(context, "projectX2");
+		ProjectCopyTask task = new ProjectCopyTask(null, "projectX2");
+
+		task.task(project.getDirectory(), project2.getName());
+
+		project2 = XstreamSerializer.getInstance().loadProject(project2.getDirectory(), context);
+		project2.getUserVariable(userVariable.getName()).setValue("false");
+
+		UserVariable userVariableProject2 = project2.getUserVariable(userVariable.getName());
+		accessor = new DeviceVariableAccessor(project2.getDirectory());
+		accessor.readUserVariableValue(userVariableProject2);
+
+		cleanupList.add(project2.getName());
+		assertEquals(userVariable.getDeviceValueKey(), userVariableProject2.getDeviceValueKey());
+		assertEquals(userVariable.getValue(), userVariableProject2.getValue());
+	}
+
+	@Test
+	public void deleteLocaleVariableTest() throws IOException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+		accessor.writeVariable(localUserVariable);
+		accessor.writeVariable(userVariable);
+		accessor.deleteAllLocalVariables(sprite);
+
+		assertFalse(accessor.readUserVariableValue(localUserVariable));
+		assertTrue(accessor.readUserVariableValue(userVariable));
+	}
+
+	@Test
+	public void cloneSpriteTest() throws IOException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+		Sprite clone = new SpriteController().copyForCloneBrick(sprite);
+
+		accessor.writeVariable(localUserVariable);
+
+		UserVariable clonedVar = clone.getUserVariable(localUserVariable.getName());
+		clonedVar.setValue("false");
+		assertTrue(accessor.readUserVariableValue(clonedVar));
+		assertEquals(localUserVariable.getValue(), clonedVar.getValue());
+	}
+
+	@Test
+	public void deleteUserVariableTest() throws IOException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+		accessor.writeVariable(userVariable);
+		assertTrue(accessor.readUserVariableValue(userVariable));
+		accessor.removeDeviceValue(userVariable);
+		assertFalse(accessor.readUserVariableValue(userVariable));
+	}
+
+	@After
+	public void tearDown() throws IOException {
+		for (String name: cleanupList) {
+			TestUtils.deleteProjects(name);
+		}
+	}
+
+	private Project createSimpleProject(String name) throws IOException {
+		DefaultProjectCreator creator = new DefaultProjectCreator();
```

Comment:
why use the default project creator? this is time intesive and builds a project with pictures, sounds, etc and all other kinds of stuff you dont need in this test.

Reply:
It didn't work with an "empty" project. And this seemed to me a easy way to create a smal project.
But ok, i will remove this creator.